include/multiple_object_tracking/clustering.hpp

```diff
@@ -211,8 +211,9 @@ auto make_point(const std::variant<Alternatives...> & detection) -> Point
 
 }  // namespace detail
 
-template <typename Detection>
-[[nodiscard]] auto cluster_detections(std::vector<Detection> detections, double distance_threshold)
+template <typename Detection, typename Metric>
```

Comment:
If you are building in some kind of customization point here it might be worth while documenting how someone would bring their own `Metrc` here

Reply:
The whole library needs some documentation overhaul. I can do that now piece-by-piece, or wait until the rest of the current PRs get merged. Do you have a preference?

Reply:
I would say that if possible piece by piece. Ideally each PR would only add documentation related to the pieces of code it touches. We can add some documentation related to how metric can be used here and save other improvements for later. Otherwise I feel that it will be deprioritized.

If you feel that it may be harder to do the documentation piece by piece then maybe we can consider some way to better divide it.

Reply:
@adamlm  this is good. I was more thinking along the lines of how someone would replace our current metric with a difference custom metric since that is really the change made available by this PR. I am not sure that is currently clear and providing this customization is a lot more valuable with information about how to use it.